integration-tests/src/test/groovy/com/okta/sdk/tests/it/GroupRolesIT.groovy

```diff
@@ -0,0 +1,213 @@
+
+        assertRolePresent(roles, userAdminRole)
+        assertRolePresent(roles, appAdminRole)
+    }
+
+    @Test
+    @Scenario("unassigned-roles-for-group")
+    void unassignRoleForGroupTest() {
+
+        String groupName = "sdk-java List Test Group ${uniqueTestName}"
+
+        // 1. Create a new group
+        Group createdGroup = GroupBuilder.instance()
+            .setName(groupName)
+            .buildAndCreate(client)
+        registerForCleanup(createdGroup)
+
+        validateGroup(createdGroup, groupName)
+
+        // 2. List all groups and find the group created
+        assertGroupPresent(client.listGroups(), createdGroup)
+
+        // 3. Create a USER_ADMIN role request
+        AssignRoleRequest userAdminRoleRequest = client.instantiate(AssignRoleRequest)
+        userAdminRoleRequest.setType(RoleType.USER_ADMIN)
+
+        // 4. Assign the above role
+        Role userAdminRole = createdGroup.assignRole(userAdminRoleRequest)
+
+        // 5. List assigned group roles and check the assignments
+        assertRolePresent(client.listGroupAssignedRoles(createdGroup.getId()), userAdminRole)
+
+        // 6. Remove the role from group
+        client.removeRoleFromGroup(createdGroup.getId(), userAdminRole.getId())
+
+        // 7. List assigned group roles and check the assignments
+        assertRoleAbsent(client.listGroupAssignedRoles(createdGroup.getId()), userAdminRole)
+    }
+
+    @Test
+    @Scenario("list-group-targets-for-group")
+    void listGroupTargetsForGroupTest() {
+
+        String groupName1 = "sdk-java List Test Group ${uniqueTestName}-1"
+        String groupName2 = "sdk-java List Test Group ${uniqueTestName}-2"
+
+        // 1. Create two groups
+        Group createdGroup1 = GroupBuilder.instance()
+            .setName(groupName1)
+            .buildAndCreate(client)
+        Group createdGroup2 = GroupBuilder.instance()
+            .setName(groupName2)
+            .buildAndCreate(client)
+        registerForCleanup(createdGroup1)
+        registerForCleanup(createdGroup2)
+
+        validateGroup(createdGroup1, groupName1)
+        validateGroup(createdGroup2, groupName2)
+
+        // 2. List all groups and find the groups created
+        assertGroupPresent(client.listGroups(), createdGroup1)
+        assertGroupPresent(client.listGroups(), createdGroup2)
+
+        // 3. Create a USER_ADMIN role request
+        AssignRoleRequest userAdminRoleRequest = client.instantiate(AssignRoleRequest)
+        userAdminRoleRequest.setType(RoleType.USER_ADMIN)
+
+        // 4. Assign the above role to the first group
+        Role userAdminRole = createdGroup1.assignRole(userAdminRoleRequest)
+
+        // 5. Add second group as the admin target
+        String href = "/api/v1/groups/" + createdGroup1.getId() + "/roles/" + userAdminRole.getId() +
+            "/targets/groups/" + createdGroup2.getId() + ""
+        userAdminRole.setResourceHref(href) //TODO: find the right way of doing this!
```

Comment:
Why do you need to modify the href manually?

Reply:
There's a method called `addGroupTargetToGroupAdministratorRoleForGroup`, is that the one you want to use here?

Reply:
Yeah, looks like it.